netstate/auth/process_api_key.go

```diff
@@ -0,0 +1,43 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package auth
+
+import (
+	"crypto/subtle"
+	"net/http"
+
+	"github.com/spf13/viper"
+)
+
+// InitializeHeaders : mocks HTTP headers to preset X-API-Key
+func InitializeHeaders() *http.Header {
+
+	httpHeaders := http.Header{
+		"Accept-Encoding": {"gzip, deflate"},
+		"Accept-Language": {"en-US,en;q=0.9"},
+		"X-Api-Key":       {"12345"},
+		"Cache-Control":   {"max-age=0"},
+		"Accept":          {"text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8"},
+		"Connection":      {"keep-alive"},
+	}
+	return &httpHeaders
+}
+
+// ValidateAPIKey : validates the X-API-Key header to an env/flag input
+func ValidateAPIKey(header string) bool {
```

Comment:
wonder if it would be useful to return an error informing of what was invalid

Reply:
when I integrate with nat's stuff, I'll have that